### PR TITLE
[Memory tests] Add new tests

### PR DESCRIPTION
**Pull request by just-sparta**
### Tickets:
 - *51074*

### CODE REVIEW

File: tests/memory_tests/include/memory_tests_helper/utils.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+* @brief Get extension from filename
+* @param filename - name of the file which extension should be extracted
+* @return string with extracted file extension
+*/
+std::string fileExt(const std::string& filename) {
+    auto pos = filename.rfind('.');
+    if (pos == std::string::npos) return "";
+    return filename.substr(pos + 1);
+}
+}
```
Comment (ishariko) [inline tests/memory_tests/include/memory_tests_helper/utils.h:20]: please add empty line in the end

File: tests/memory_tests/src/memory_tests_helper/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+file (GLOB SRC *.cpp)
+add_library(${TARGET_NAME} STATIC ${SRC})
+target_include_directories(${TARGET_NAME} PUBLIC "${CMAKE_SOURCE_DIR}/include")
+
+add_subdirectory(${OpenVINO_SOURCE_DIR}/thirdparty/gflags
+                 ${CMAKE_CURRENT_BINARY_DIR}/gflags_build
+                 EXCLUDE_FROM_ALL)
+
+target_link_libraries(${TARGET_NAME} PUBLIC gflags)
```
Comment (ishariko) [inline tests/memory_tests/src/memory_tests_helper/CMakeLists.txt:15]: empty line

File: tests/memory_tests/test_runner/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+pytest~=5.0
+PyYAML==5.4.1
+jsonschema==3.2.0
+distro==1.5.0
+numpy>=1.19.2
+pymongo
```
Comment (ishariko) [inline tests/memory_tests/test_runner/requirements.txt:6]: Please add version for python modules

**Review by akoryach — APPROVED** [review]

File: tests/memory_tests/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+2. Install tests:
+``` bash
+make install tests
+```
+
+3. Run test:
+``` bash
+./scripts/run_memorytest.py tests/memtest_infer -m model.xml -d CPU
```
[the remark this answers is not in the document]
Comment (vurusovs) [reply tests/memory_tests/README.md:36]: Also note that scripts should be run from <install_dir>, not from repo tree

**Review by ishariko — APPROVED** [review]

**Review by vurusovs — APPROVED** [review]

**Review by vurusovs — APPROVED** [review]

**Pull request merged**